Atomic/syscontainers.py

```diff
@@ -292,7 +292,7 @@ def _checkout(self, repo, name, img, deployment, upgrade, values=None, destinati
             return self._do_checkout(repo, name, img, upgrade, values, destination, unitfileout, tmpfilesout, extract_only, remote)
         except (ValueError, OSError) as e:
             try:
-                if not extract_only:
+                if not upgrade and not extract_only:
```

Comment:
nit, just to remain consistent with the lines below, should this be "if not extract_only and not upgrade:"